src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -129,8 +129,8 @@ private static String getHash(@CheckForNull SCMRevision revision) {
             // unwrap
             revision = ((PullRequestSCMRevision) revision).getPull();
         }
-        if (revision instanceof MercurialSCMSource.MercurialRevision) {
-            return ((MercurialSCMSource.MercurialRevision) revision).getHash();
+        if (revision instanceof BitbucketSCMSource.MercurialRevision) {
+            return ((BitbucketSCMSource.MercurialRevision) revision).getHash();
```

Comment:
Maybe to be on the safe side add a else if for `MercurialSCMSource.MercurialRevision` ?